### PR TITLE
Implement Wayland Support for different applications

### PR DESCRIPTION
**Pull request by NextWork123**

### CODE REVIEW

**Comment by SoongVilda** [comment]
@ptr1337 What do you think about these changes to "force" Wayland? Do you have experiences with it? As for the me, I only used some of these entries, but not all.

**Comment by ptr1337** [comment]
> @ptr1337 What do you think about these changes to "force" Wayland? Do you have experiences with it? As for the me, I only used some of these entries, but not all.

Im testing this two more days. So far the latest commit works fine.
This should be also added to the zsh config please.

**Comment by NextWork123** [comment]
I will do the commit too for zsh. Me and @vnepogodin choose those flags. Let me know if I need to change something.

**Comment by ptr1337** [comment]
@NextWork123 

So far with the current flags no issues on nvidia wayland. So it was probably due the commit before the issues I had. Give this a bit more days testing, then we can merge it

**Comment by NextWork123** [comment]
Yeah, pick your time. I'm testing as well. Only faced issue about the last flag I unset for EAC Games.

File: config.fish [hunk 1]
```diff
@@ -13,9 +13,19 @@ end
 set -U __done_min_cmd_duration 10000
 set -U __done_notification_urgency_level low
 
-## Enable Wayland support for Mozilla app if session type is Wayland
+## Enable Wayland support for different applications
 if [ "$XDG_SESSION_TYPE" = "wayland" ]
+    set -gx WAYLAND 1
+    set -gx QT_QPA_PLATFORM 'wayland;xcb'
+    set -gx GDK_BACKEND wayland
```
Comment (ventureoo) [inline config.fish:20]: That should be the default, no?

Comment (NextWork123) [reply config.fish:20]: Better to set anyway the flag for only wayland, sometimes it doesn't put that flags at automatically.

File: config.fish [hunk 2]
```diff
@@ -13,9 +13,19 @@ end
...
+    set -gx MOZ_DBUS_REMOTE 1
     set -gx MOZ_ENABLE_WAYLAND 1
+    set -gx _JAVA_AWT_WM_NONREPARENTING 1
+    set -gx BEMENU_BACKEND wayland
+    set -gx CLUTTER_BACKEND wayland
+    set -gx ECORE_EVAS_ENGINE wayland_egl
+    set -gx ELM_ENGINE wayland_egl
+    set -gx SAL_USE_VCLPLUGIN gtk3
```
Comment (ventureoo) [inline config.fish:28]: Will this break the KF plugin for libreoffice in the Wayland Plasma session?

Comment (NextWork123) [reply config.fish:28]: i don't think they gonna break the KF plugin,but test and let us know.

Comment (vnepogodin) [reply config.fish:28]: Np. that will use GTK+ file dialog

Comment (ventureoo) [reply config.fish:28]: It may be worth force gtk3 if $XDG_CURRENT_DESKTOP is not set to KDE.

Comment (NextWork123) [reply config.fish:28]: if you set instead of gtk3, kf5? the SAL_USE_VCLPLUGIN is working again fine?

**Comment by ventureoo** [comment]
Works well for me now. Although you could probably add ``EGL_PLATFORM=wayland``

**Comment by NextWork123** [comment]
Ok, let me test that flag aswell on amd gpu. if is all right i gonna do the commit for add :+1:

**Comment by ptr1337** [comment]
At me is also everything fine.

Lets merge it.  Thank you for the PR

**Pull request merged**